src/CapabilitiesUtil/CapabilitiesUtil.js

```diff
@@ -88,41 +84,43 @@ class CapabilitiesUtil {
    * @param {Object} capabilities A capabilities object.
    * @param {string} nameField Configure the field which should be set as the
    *                           'name' property in the openlayers layer.
-   * @param {Function} proxyFn Optional proxy function which can be applied to
+   * @param {(url: string) => string} proxyFn Optional proxy function which can be applied to
    *                           `GetMap`, `GetFeatureInfo` and `GetLegendGraphic`
    *                           requests to avoid CORS issues.
-   * @return {OlLayerTile[]} Array of OlLayerTile
+   * @return {import("ol/layer/Tile").default[]} Array of OlLayerTile
    */
   static getLayersFromWmsCapabilities(capabilities, nameField = 'Name', proxyFn) {
-    const wmsVersion = get(capabilities, 'version');
-    const layersInCapabilities = get(capabilities, 'Capability.Layer.Layer');
-    const wmsGetMapConfig = get(capabilities, 'Capability.Request.GetMap');
-    const wmsGetFeatureInfoConfig = get(capabilities, 'Capability.Request.GetFeatureInfo');
-    const getMapUrl = get(wmsGetMapConfig, 'DCPType[0].HTTP.Get.OnlineResource');
-    const getFeatureInfoUrl = get(wmsGetFeatureInfoConfig, 'DCPType[0].HTTP.Get.OnlineResource');
+    const wmsVersion = _get(capabilities, 'version');
+    const layersInCapabilities = _get(capabilities, 'Capability.Layer.Layer');
+    const wmsGetMapConfig = _get(capabilities, 'Capability.Request.GetMap');
+    const wmsGetFeatureInfoConfig = _get(capabilities, 'Capability.Request.GetFeatureInfo');
+    const getMapUrl = _get(wmsGetMapConfig, 'DCPType[0].HTTP.Get.OnlineResource');
+    const getFeatureInfoUrl = _get(wmsGetFeatureInfoConfig, 'DCPType[0].HTTP.Get.OnlineResource');
     const legendUrl = layersInCapabilities.length > 0
-      ? get(layersInCapabilities[0], 'Style[0].LegendURL[0].OnlineResource')
+      ? _get(layersInCapabilities[0], 'Style[0].LegendURL[0].OnlineResource')
       : null;
 
     return layersInCapabilities.map(layerObj => {
-      const title = get(layerObj, 'Attribution.Title');
-      const onlineResource = get(layerObj, 'Attribution.OnlineResource');
+      const title = _get(layerObj, 'Attribution.Title');
+      const onlineResource = _get(layerObj, 'Attribution.OnlineResource');
       const attributions = [onlineResource ? `<a target="_blank" href="${onlineResource}">${title}</a>` : title];
 
       return new OlLayerImage({
         opacity: 1,
-        title: get(layerObj, 'Title'),
-        name: get(layerObj, nameField),
-        abstract: get(layerObj, 'Abstract'),
-        getFeatureInfoUrl: isFunction(proxyFn) ? proxyFn(getFeatureInfoUrl) : getFeatureInfoUrl,
-        getFeatureInfoFormats: get(wmsGetFeatureInfoConfig, 'Format'),
-        legendUrl: isFunction(proxyFn) ? proxyFn(legendUrl) : legendUrl,
-        queryable: get(layerObj, 'queryable'),
+        properties: {
```

Comment:
Also a quite useful but substantial change, should be documented.

Reply:
this will result in the same object. The `properties` in the constructor are specifically for typescript. See https://openlayers.org/en/latest/apidoc/module-ol_layer_Layer.html

Reply:
Ok, thanks for the clarification

Reply:
Or in the code here: https://github.com/openlayers/openlayers/blob/v6.13.0/src/ol/layer/Base.js#L98 :)